lib/Mojo/Cache.pm

```diff
@@ -0,0 +1,73 @@
+package Mojo::Cache;
+use Mojo::Base -base;
+
+has 'max_keys' => 100;
+
+sub get { (shift->{cache} // {})->{shift()} }
+
+sub set {
+  my ($self, $key, $value) = @_;
+
+  return $self unless (my $max = $self->max_keys) > 0;
+
+  my $cache = $self->{cache} //= {};
+  my $queue = $self->{queue} //= [];
+  delete $cache->{shift @$queue} while @$queue >= $max;
+  push @$queue, $key unless exists $cache->{$key};
+  $cache->{$key} = $value;
+
+  return $self;
+}
+
+1;
+
+=encoding utf8
+
+=head1 NAME
+
+Mojo::Cache - Naive in-memory cache
```

Comment:
Mojo::Cache probably exists just for the renderer, and for each process to re-cache is probably a trivial matter.  Nevertheless, this module does it exist and, like Mojo::Collection and others, it's just asking to be used.  However, in some use cases it probably should not be used if using prefork.  Can this warning be documented somewhere?  And can an alternate suggestion or pointer be given as to what to do instead?  Is a database the only way?  What about Mojo::MemoryMap?  Is that a good use-case?